protostar/commands/test/test_suite.py

```diff
@@ -3,8 +3,16 @@
 from typing import List, Optional
 
 
+@dataclass(frozen=True)
+class TestCase:
+    test_fn_name: str
+
+
 @dataclass(frozen=True)
 class TestSuite:
     test_path: Path
-    test_case_names: List[str]
+    test_cases: List[TestCase]
     setup_fn_name: Optional[str] = None
+
+    def get_test_case_names(self) -> List[str]:
+        return [tc.test_fn_name for tc in self.test_cases]
```

Comment:
```suggestion
    @property
    def test_case_names(self) -> List[str]:
        return [tc.test_fn_name for tc in self.test_cases]
```

Reply:
this is non-trivial operation (its constructing a new list each call), hence I used a method approach

Reply:
maybe `collect_test_case_names`?

Reply:
Done.